### ISSUE TITLE
Fix S1144: Issues is raised while method is being used (DebuggerDisplayAttribute)

### ISSUE THREAD

**Issue by Evangelink** [SonarSource/sonar-csharp#1195]
Copy-paste from https://github.com/SonarSource/sonarlint-visualstudio/issues/571

### Description

There are moments when .Net uses reflection as the primary access point for interaction.  Sometimes it is beneficial to take advantage of this pattern, but that is neither here nor there.  I am not sure if it is even possible to track this, but might as well mention it.
When using reflection for class member access it can sometimes appear that a method/properties is never referenced.   (RSPEC-1144)

### Repro steps

1. DebuggerDisplayAttribute
```
    [DebuggerDisplay("{DebugDisplay()}")]
    public class Foo
    {
        public string Bar { get; set; }

        string DebugDisplay()
        {
            return $"Bar: {this.Bar}";
        }
    }

```
2. Build with SonarAnalyzer - Appears in Messages (not warnings)

### Expected behavior

Should not be flagged as "unreferenced"

### Actual behavior

warning S1144: Remove the unused private method 'DebugDisplay'.

### Known workarounds

Ignore it.

### Related information

* SonarLint for VS Version: 6.8.2.4717
* Visual Studio Version: 14.0.25424.00 UPdate 3
* SonarQube server version: 5.6.7 
* Languages in your solution: C#
